crates/trie/trie/src/proof_v2/mod.rs

```diff
@@ -599,6 +1092,12 @@ where
         #[cfg(not(debug_assertions))]
         let targets = targets.into_iter();
 
+        // Convert B256 targets into Nibbles.
+        let targets = targets.into_iter().map(|key| {
+            // SAFETY: key is a B256 and so is exactly 32-bytes.
+            unsafe { Nibbles::unpack_unchecked(key.as_slice()) }
```

Comment:
I feel like we should add to `nybbles` a safe function for unpacking `[u8; 32]`, basically a full path